### PR TITLE
Andy/race next available

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
Demonstrates race condition in next-available-prefix API

### CODE REVIEW

File: nautobot/core/templates/nautobot_config.py.j2 [hunk 1]
```diff
@@ -75,6 +75,17 @@ RQ_QUEUES = {
 # For detailed configuration see: https://github.com/Suor/django-cacheops#setup
 CACHEOPS_REDIS = "redis://localhost:6379/1"
 
+# This is used for configuring Redis locks via django caching backend.
+CACHES = {
+    "default": {
+        "BACKEND": "django_redis.cache.RedisCache",
+        "LOCATION": "redis://127.0.0.1:6379/1",
```
Comment (glennmatthews) [inline nautobot/core/templates/nautobot_config.py.j2:82]: Do we need to use the IP address here, or can we change it to `localhost` to match the `CACHEOPS_REDIS` config above?

**Review by glennmatthews — COMMENTED** [review]
We can remove `django-pglocks` from the project dependencies as part of this PR, correct?

File: nautobot/core/templates/nautobot_config.py.j2 (hunk 1 above)
Comment (andrew-wm-arthur) [reply nautobot/core/templates/nautobot_config.py.j2:82]: @jathanism had some thoughts here

File: pyproject.toml [hunk 2]
```diff
@@ -87,6 +87,7 @@ PyYAML = "~5.4.1"
 social-auth-app-django = "^4.0.0"
 # Rendering of SVG images (for rack elevations, etc.)
 svgwrite = "~1.4.1"
+django-redis = "^4.12.1"
```
Comment (jathanism) [inline pyproject.toml:88]: Can you please alpha sort this in this section of the file and add a comment above it? (I suggested a comment)

```suggestion
# Redis cache for Django used for distributed locking
django-redis = "^4.12.1"
```

File: nautobot/ipam/api/views.py [hunk 3]
```diff
@@ -22,6 +21,7 @@
     VLANGroup,
     VRF,
 )
+from nautobot.utilities.constants import ADVISORY_LOCK_KEYS
```
Comment (jathanism) [inline nautobot/ipam/api/views.py]: We can ditch this and the associated `ADVISORY_LOCK_KEYS` constant.

**Review by jathanism — CHANGES_REQUESTED** [review]
We'll still need the config template and docs to be updated for the new `CACHES` setting. I'm happy to help with those. I suggested a couple minor tweaks in the interim.

File: nautobot/core/templates/nautobot_config.py.j2 (hunk 1 above)
Comment (jathanism) [reply nautobot/core/templates/nautobot_config.py.j2:82]: My thought was that not only can we use localhost, but we could use the exact same URI for both `CACHEOPS_REDIS` and `CACHES["default"]["LOCATION"]`.

One thing to consider, however, is the possible confusion of `django-redis` cache settings vs. `django-cacheops` which does caching, but not using Django's cache framework. 

Another important consideration is the "what if" of using the same Redis instance/db for both and someone invalidates the Cacheops cache using `nautobot-server invalidate all` (which flushes the Redis db) and nukes a write lock in the process.

**Review by jathanism — APPROVED** [review]

**Pull request merged**